File: Voticus.Api/Domain/Poll.cs
namespace Voticus.Api.Domain;

public class Poll
{
    public int Id { get; set; }
    public string Title { get; set; } = default!;
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
    public DateTime? FinishAtUtc { get; set; }
    public ICollection<Entry> Entries { get; set; } = new List<Entry>();
}

File: Voticus.Client/src/types/poll.ts
// src/types/poll.ts

export interface EntryDto {
  id: number;
  pollId: number;
  description: string;
  imageUrl?: string;
  voteCount: number; // not optional now
}

export interface PollDto {
  id: number;
  title: string;
  description?: string;
  imageUrl?: string;
  finishAtUtc?: string; // ISO string from backend
  entries?: EntryDto[]; // populated in GET /api/polls/{id}
}
